torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -285,16 +280,31 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                 raise NotImplementedError(
                     f"FSDP only supports 1D TP, not {self._tp_spec.placements}"
                 )
-            # TODO: Hard code FSDP + TP; need to support HSDP + TP
             split_factor = self._tp_spec.num_shards_map[0]
-            self._spmd_placements: Tuple[Placement, ...] = (
-                (
-                    _StridedShard(0, split_factor=split_factor)
-                    if split_factor > 1
-                    else Shard(0)
-                ),
-                self._tp_spec.placements[0],
-            )
+            assert (
+                2 <= self._spmd_mesh.ndim <= 3
+            ), f"_spmd_mesh.ndim can only be 2 or 3 but got {self._spmd_mesh.ndim}."
+            self._spmd_placements: Tuple[Placement, ...]
+            if self._spmd_mesh.ndim == 2:
```

Comment:
nit: maybe factor out to avoid some duplication?

```
fsdp_shard_placement = (
    _StridedShard(0, split_factor=split_factor)
    if split_factor > 1
    else Shard(0)
)
if self._spmd_mesh.ndim == 2:
    self._spmd_placements = (fsdp_shard_placement, self._tp_spec.placements[0])
else:
    self._spmd_placements = (
        Replicate(), fsdp_shard_placement, self._tp_spec.placements[0])
    )
```